### DISCUSSION TITLE
Add newline in Dialog Prompt

### DISCUSSION THREAD

**Discussion by cryptoKevinL** [MetaMask/snaps#1480] [Q&A] [answered]
I can't seem to get a newline into a snaps Dialog Prompt, should this work or do I have to use a new text() object in the panel for anytime I want a newline?  my code:

```js
Object.values(chatHistory).forEach(val => {
  if(address.toLowerCase() === val.fromaddr.toLowerCase()) { 
    convoBody += ' **' + from + ':** ' + val.message + '\n'
  } else {
    convoBody += ' **me:** ' + val.message + '\n'
  }
});
...
const diagResponse = await snap.request({
    method: 'snap_dialog',
    params: {
      type: 'prompt',
      content: panel([heading('New Message From: ' + from), text(convoBody)]),
      placeholder: 'Enter response to message here...',
    },
});
```

my result: 
<img width="342" alt="Screenshot 2023-06-07 at 9 26 06 PM" src="https://github.com/MetaMask/snaps/assets/19207330/b8387a5a-c8fc-43e4-a4cd-69b99a6ddfe8">

I wanted to build my panel programmatically, hopefully it can be done still?  Maybe I just need to build the union of objects to put in the panel() call, but I tried a few times and my syntax was bad maybe.

**Comment by Montoya** [comment, marked as answer]
There's no newline character, you would have to use a new `text()` container for each line

**Reply by cryptoKevinL** [reply]
Thanks, do you have an example or do you know how to add `text()` containers in pieces to an object to pass to `panel()`?

I tried things like this without success:
```
let objContainer = [text("hello")]
objContainer.push(text("world"))
...
content: panel(objContainer),
```

**Reply by ziad-saab** [reply]
Yes, this is exactly how you would do it. Can you link to your full code on GitHub?

**Reply by Montoya** [reply]
Just FYI, here's an example code from the MobyMask snap: 

```
export const onTransaction: OnTransactionHandler = async ({ transaction }) => {
  const insights = await getInsights(transaction); 
  const titles = Object.keys(insights); 
  let arr = []; 
  for(var i=0;i<titles.length;i++) { 
    arr.push(heading(titles[i])); 
    arr.push(text(insights[titles[i]])); 
  }

  return {
    content: panel(arr)
  };
};
```

You should be able to push to an array and then put that in a panel...

**Reply by cryptoKevinL** [reply]
ok, it works now.   I think I was just tired last night :)
<img width="351" alt="Screenshot 2023-06-08 at 1 18 21 PM" src="https://github.com/MetaMask/snaps/assets/19207330/1793dd86-be04-46e5-b997-41c44c4fa3da">

Last question, any way to auto scroll to bottom in a dialog prompt?  We might just show the last 5 or 6 messages so this may be a non-issue, but wanted to ask in case.
